decentraland/components/stateful/EthereumUpgradeSecurity/EthereumUpgradeSecurity.tsx

```diff
@@ -2,38 +2,330 @@ import React, { useState, useEffect } from "react";
+//   const safeContract = await getGnosisSafeInstanceAt(safeAddress)
+//   const safeProps = await buildSafe(safeAddress, name)
+//   const owners = getOwnersFrom(ownersNames, ownerAddresses)
+//   safeProps.owners = owners
+
+//   addSafe(safeProps)
+
+//   // returning info for testing purposes, in app is fully async
+//   return { safeAddress: safeContract.address, safeTx: safe }
+// }
+
+interface EthereumUpgradeSecurityProps {
+  account: any;
+}
+
+const EthereumUpgradeSecurity: React.FunctionComponent<EthereumUpgradeSecurityProps> = ({
+  account,
+}) => {
   const [isDeployed, setIsDeployed] = useState(null);
+  const [safeAddress, setSafeAddress] = useState(null);
+
+  const { signingKey } = account;
+  const { address: ephemeralAccountAddress } = signingKey;
+
+  console.log({ ephemeralAccountAddress });
 
   useEffect(() => {
-    function handleDeploymentComplete() {
-      setIsDeployed(true);
-    }
+    const onCallSafeContractSubmit = async () => {
+      try {
+        // TODO: Get userAccount dynamically from redux
+        // Direct contract deployment version:
+        // createSafe(values, userAccount, addSafe)
+
+        // const userAccount = "0x9b1c8920659cd7db3dee1ae55ec1dc7d0e500c96";
+        // const checksummedUserAccount = ethers.utils.getAddress(userAccount);
+
+        //
+        // Enable and disable this section to avoid hitting
+        // Gnosis Safe relay service API too much
+        // ---
+        if (!isDeployed) {
+          const checksummedUserAccount = ephemeralAccountAddress;
+          const safeAddress = await postSafe([checksummedUserAccount], 1);
+          console.log({ safeAddress });
+          // TODO: Eventually we won't want to send this from a
+          // hardcoded funding source
+          console.log("About to fund with ethers");
+          // await fundWithEthers(safeAddress);
+          await fundWithEthersVanilla(safeAddress);
+          console.log("Done funding contract with ethers");
+
+          // Optional: Hit the /put address to signal it has been funded
+          await putFunded(safeAddress);
+
+          // TODO: Confirm contract successfully deployed
+
+          setSafeAddress(safeAddress);
+          setIsDeployed(true);
+        } else {
+          console.log("Already deployed");
+        }
+
+        // ---
+        // End section to enable and disable
+        //
+      } catch (error) {
+        // eslint-disable-next-line
+        console.error("Error while creating the Safe: " + error);
+      }
+    };
+
+    console.log("EthereumUpgradeSecurity useEffect");
 
     // TODO: Hit the API to deploy a new contract
 
-    handleDeploymentComplete();
+    onCallSafeContractSubmit();
 
     return function cleanup() {
       // TODO: Cancel any unwanted remnants of having already made this request
     };
-  });
+  }, [account]); // Only re-run the effect if account changes
+
+  // TODO: Show loading by setting a boolean or enum value when deployment starts
```

Comment:
Maybe we could use a state hook for that to set loading value true/false inside `onCallSafeContractSubmit` try/catch block

Reply:
@pedrobergamini Yep, I think that sounds good!

Reply:
How would that handle persistence if someone quits the app and reopens it during the process?

Reply:
I created an issue to track this improvement. https://github.com/tasitlabs/tasit/issues/478